test/e2e/node_problem_detector.go

```diff
@@ -193,18 +193,16 @@ var _ = framework.KubeDescribe("NodeProblemDetector", func() {
 			Expect(f.WaitForPodRunning(name)).To(Succeed())
 		})
 
-		It("should generate default node condition after started", func() {
+		It("should generate node condition and events for corresponding errors", func() {
 			By("Make sure no events are generated")
 			Consistently(func() error {
 				return verifyNoEvents(c.Events(eventNamespace), eventListOptions)
 			}, pollConsistent, pollInterval).Should(Succeed())
-			By("Make sure the node condition is false")
+			By("Make sure the default node condition is false")
 			Consistently(func() error {
 				return verifyCondition(c.Nodes(), node.Name, condition, api.ConditionFalse, defaultReason, defaultMessage)
 			}, pollConsistent, pollInterval).Should(Succeed())
-		})
 
-		It("should generate event when temporary error happens", func() {
 			num := 3
 			By(fmt.Sprintf("Inject %d temporary errors", num))
```

Comment:
Can we leave the temporary case out of this single block? Otherwise, we won't test the concurrent updates on a single node object, right?